Sources/Clappr_iOS/Classes/Plugin/Core/Drawer/DrawerPlugin.swift

```diff
@@ -0,0 +1,82 @@
+open class DrawerPlugin: OverlayPlugin {
+    public enum Position {
+        case undefined
+    }
+
+    open var position: DrawerPlugin.Position {
+        return .undefined
+    }
+
+    open var size: CGSize {
+        return .zero
+    }
+
+    open var placeholder: CGFloat {
+        return .zero
+    }
+
+    private(set) var isClosed: Bool = true {
+        willSet {
+            let event: Event = newValue ? .willHideDrawerPlugin : .willShowDrawerPlugin
+            core?.trigger(event)
+        }
+        didSet {
+            let event: Event = isClosed ? .didHideDrawerPlugin : .didShowDrawerPlugin
+            core?.trigger(event)
+        }
+    }
+
+    public required init(context: UIObject) {
+        super.init(context: context)
+        view.alpha = .zero
+    }
+
+    open override func bindEvents() {
+        bindMediaControlEvents()
+        bindDrawerEvents()
+    }
+
+    private func bindMediaControlEvents() {
+        guard let core = core else { return }
+
+        listenTo(core, event: .willShowMediaControl) { [weak self] _ in
+            UIView.animate(withDuration: ClapprAnimationDuration.mediaControlShow) {
+                self?.view.alpha = 1
```

Comment:
Do we have to check if drawer is open before make it visible?

Reply:
The `isClosed` state stands for wether it should keep visible if the mediaControl is going to be **dismissed**. In this case, we always want the drawer to be visible when mediaControl is going to be presented. Therefore, we don't have to implement such verification.